crates/pipeline-manager/src/api/endpoints/pipeline_interaction.rs

```diff
@@ -783,7 +837,99 @@ pub(crate) async fn get_pipeline_circuit_json_profile(
             request.query_string(),
             Some(Duration::from_secs(120)),
         )
+        .await?;
+
+    // If response is not successful, return it as-is
+    let status = response.status();
+    if !status.is_success() {
+        return Ok(response);
+    }
+
+    // Extract the JSON body
+    let body_bytes = actix_web::body::to_bytes(response.into_body())
+        .await
+        .map_err(|e| {
+            ManagerError::from(ApiError::UnableToCreateSupportBundle {
+                reason: format!("Failed to read dump_json_profile response body: {}", e),
+            })
+        })?;
+
+    // Create a ZIP file containing the JSON profile
+    let zip_bytes = create_zip([("profile.json", body_bytes.as_ref())])?;
```

Comment:
since this is just one file, I think we should just compress all requests as a middleware https://actix.rs/docs/response/#content-encoding